### PR TITLE
Tests for Recommend Classes 

### PR DESCRIPTION
**Pull request by skellamp**

### CODE REVIEW

File: module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SummonResultsTest.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class SummonResultsTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Test getting search class id.
+     *
+     * @return void
+     */
+    public function testGetResults(): void
```
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SummonResultsTest.php]: I'd suggest renaming this method to `testGetSearchClassId()`.

File: module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SummonBestBetsTest.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            ->disableOriginalConstructor()
+            ->getMock();
+        $results->expects($this->once())->method('getParams')
+            ->will($this->returnValue($parms));
+        $parms->expects($this->once())->method('getSearchClassId')
+            ->will($this->returnValue("Summon"));
+        $obj->process($results);
+        $results->expects($this->once())->method('getBestBets')
+            ->will($this->returnValue(false));
+        $this->assertFalse($obj->getresults());
```
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SummonBestBetsTest.php]: `getresults` should be `getResults` (PHP function names are case-insensitive, but we try to be consistent as much as possible).

File: module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SummonDatabasesTest.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            ->disableOriginalConstructor()
+            ->getMock();
+        $results->expects($this->once())->method('getParams')
+            ->will($this->returnValue($parms));
+        $parms->expects($this->once())->method('getSearchClassId')
+            ->will($this->returnValue("Summon"));
+        $obj->process($results);
+        $results->expects($this->once())->method('getDatabaseRecommendations')
+            ->will($this->returnValue(false));
+        $this->assertFalse($obj->getresults());
```
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SummonDatabasesTest.php]: Another `getresults` that should be `getResults`.

File: module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SummonTopicsTest.php [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            ->disableOriginalConstructor()
+            ->getMock();
+        $results->expects($this->once())->method('getParams')
+            ->will($this->returnValue($parms));
+        $parms->expects($this->once())->method('getSearchClassId')
+            ->will($this->returnValue("Summon"));
+        $obj->process($results);
+        $results->expects($this->once())->method('getTopicRecommendations')
+            ->will($this->returnValue(false));
+        $this->assertFalse($obj->getresults());
```
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SummonTopicsTest.php]: Another `getresults` that should be `getResults`.

File: module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SwitchTabTest.php [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+class SwitchTabTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Data provider for testGetActiveTab.
+     *
+     * @return array
+     */
+    public function tabConfigProvider(): array
+    {
+        return ['Test1' => [[["id" => "A01",
```
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SwitchTabTest.php]: Might be worth giving the tests more meaningful names -- e.g. "Test1" could be "First tab selected" and "Test2" could be "No tab selected."

File: module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SwitchTabTest.php [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        $obj = new SwitchTab();
+        $this->assertSame($expectedResult, $obj->getInactiveTabs($tabEnv));
+    }
+
+    /**
+     * Test init method.
+     *
+     * @return void
+     */
+    public function testInit(): void
```
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SwitchTabTest.php]: I'm not sure if we need tests for init, setConfig, or process, since none of them currently do anything. It's hard to come up with meaningful assertions for these. My concern is that, if logic is added to these methods in the future, it's possible that these tests could provide coverage without actually asserting anything useful, which turn out to be a problem... but if you think they're of value, I'm not insisting that we remove them. :-)

File: module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SwitchTypeTest.php [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+        $obj->process($results);
+        $this->assertSame($results, $obj->getresults());
+    }
+
+    /**
+     * Test init method.
+     *
+     * @return void
+     */
+    public function testInit(): void
```
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SwitchTypeTest.php]: Another method that might not need a test since it currently contains no logic.

File: module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SwitchTypeTest.php [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+            ->getMock();
+        $parms = $this->getMockBuilder(\VuFind\Search\Base\Params::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $results->expects($this->once())->method('getParams')
+            ->will($this->returnValue($parms));
+        $parms->expects($this->once())->method('getSearchHandler')
+            ->will($this->returnValue("bar"));
+        $obj->process($results);
+        $this->assertSame($results, $obj->getresults());
```
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/SwitchTypeTest.php]: Another all-lowercase `getresults`.

File: module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/WebResultsTest.php [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class WebResultsTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Test getting search class id.
+     *
+     * @return void
+     */
+    public function testGetResults(): void
```
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/WebResultsTest.php]: Another test name that you might want to change to testGetSearchClassId.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @skellamp, this is great progress! A few very minor comments. Once these are resolved, do you want me to merge this, or are you planning to add further tests? (I'd certainly be happy to merge sooner rather than later, and address more tests in a separate PR -- that might make the review process simpler and get the coverage improvements in place more quickly... but if you prefer a different workflow, that's fine too!)

**Review by demiankatz — APPROVED** [review]
